File: SQLiteRepository/SQLiteFluent/CommonSqlite.cs
using System;
using System.Collections.Generic;
using System.Data.SQLite;
using System.Dynamic;
using System.Linq;

namespace SQLiteRepository.SQLiteFluent
{
    public class CommonSqlite
    {

        public static SQLiteConnection CreateConnection(string path)
        {
            return new SQLiteConnection(@"Data Source=" + path + ";Version=3;");
        }

        internal static object GetWhereClause(string @where, Tuple<string,List<object>> whereIn)
        {
            if (!string.IsNullOrEmpty( where))
            {
                return " WHERE " + where;
            }
            if (whereIn != null && whereIn.Item2.Any())
            {

                return " WHERE [" + whereIn.Item1 + "] IN (" + StringUtil.Concatenate(whereIn.Item2) + ")";
            }
            return string.Empty;
        }
    }
}

File: SQLiteAdapters/CommonSqlite.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Data.SQLite


Public Class CommonSqlite

    Public Shared Function CreateConnection(path As String) As SQLiteConnection
        Return New SQLiteConnection("Data Source=" & path & ";Version=3;")
    End Function

    Friend Shared Function GetWhereClause(where As String, whereIn As Tuple(Of String, List(Of String))) As String
        If Not String.IsNullOrEmpty(where) Then
            Return " WHERE " & where
        End If
        If whereIn IsNot Nothing AndAlso whereIn.Item2.Any() Then

            Return " WHERE [" & whereIn.Item1 & "] IN (" & StringUtil.ConcatValues(whereIn.Item2, "'", "'", ",") & ")"
        End If
        Return String.Empty
    End Function

    Public Shared Function GetOrderByClause(ByVal orderBy As List(Of String)) As String
        If orderBy IsNot Nothing Then
            Return " ORDER BY " & StringUtil.ConcatValues(orderBy, "[", "]", ",")
        End If
        Return String.Empty
    End Function
End Class